blockchain/chain.go

```diff
@@ -1086,7 +1087,8 @@ func (b *BlockChain) reorganizeChainInternal(targetTip *blockNode) error {
 			err = b.checkConnectBlock(n, block, parent, view, &stxos,
 				&hdrCommitments)
 			if err != nil {
-				if _, ok := err.(RuleError); ok {
+				var rerr RuleError
+				if errors.As(err, &rerr) {
```

Comment:
Since the concrete error isn't being used, shouldn't these just be `errors.Is`?

Reply:
Nevermind.  It's checking for the type not a specific value, so this is correct as is.